### PR TITLE
Add temp API for creating scan objects

### PR DESCRIPTION
**Pull request by kholdaway**

### CODE REVIEW

File: quipucords/api/scan/model.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
+#
+# Copyright (c) 2017-2018 Red Hat, Inc.
```
Comment (chambridge) [inline quipucords/api/scan/model.py]: Did we start this in 2017

File: quipucords/api/scan/serializer.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
+#
+# Copyright (c) 2017 Red Hat, Inc.
```
Comment (chambridge) [inline quipucords/api/scan/serializer.py]: 2018

File: quipucords/api/scan/view.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
+#
+# Copyright (c) 2017-2018 Red Hat, Inc.
```
Comment (chambridge) [inline quipucords/api/scan/view.py]: Just 2018

File: quipucords/api/scan/view.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+from api.common.util import is_int
+from api.models import (Scan, Source)
+from api.serializers import (ScanSerializer)
+
+SOURCES_KEY = 'sources'
+RESULTS_KEY = 'task_results'
+TASKS_KEY = 'tasks'
+SYSTEMS_COUNT_KEY = 'systems_count'
+SYSTEMS_SCANNED_KEY = 'systems_scanned'
+SYSTEMS_FAILED_KEY = 'systems_failed'
```
Comment (chambridge) [inline quipucords/api/scan/view.py]: Don't think we need all these constants in Scan View.

File: quipucords/api/scan/view.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+
+# pylint: disable=too-many-branches
+def expand_scan(json_scan):
+    """Expand the source and calculate values.
+
+    Take scan object with source ids and pull objects from db.
```
Comment (chambridge) [inline quipucords/api/scan/view.py]: we aren't really calculating values here.

File: quipucords/api/scan/view.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    if authentication_enabled:
+        authentication_classes = (ExpiringTokenAuthentication,
+                                  SessionAuthentication)
+        permission_classes = (IsAuthenticated,)
+
+    queryset = Scan.objects.all()
+    serializer_class = ScanSerializer
+    filter_backends = (DjangoFilterBackend, OrderingFilter)
+    filter_class = ScanFilter
+    ordering_fields = ('id', 'scan_type', 'status', 'start_time', 'end_time')
```
Comment (chambridge) [inline quipucords/api/scan/view.py]: Don't think the ordering makes sense here with status, start_time, end_time.

Comment (chambridge) [reply quipucords/api/scan/view.py]: probably need ordering by name here.

File: quipucords/api/scan/view.py [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    ordering = ('id',)
```
Comment (chambridge) [inline quipucords/api/scan/view.py]: default ordering by name

File: quipucords/api/scan/view.py [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+    # pylint: disable=unused-argument
+    def create(self, request, *args, **kwargs):
+        """Create a scanjob."""
```
Comment (chambridge) [inline quipucords/api/scan/view.py]: doc strings ...

File: quipucords/api/scan/view.py [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=status.HTTP_201_CREATED,
+                        headers=headers)
+
+    def list(self, request):  # pylint: disable=unused-argument
+        """List the collection of scan jobs."""
```
Comment (chambridge) [inline quipucords/api/scan/view.py]: doc strings ...

File: quipucords/api/scan/view.py [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+        for scan in queryset:
+            serializer = ScanSerializer(scan)
+            json_scan = serializer.data
+            expand_scan(json_scan)
+            result.append(json_scan)
+        return Response(result)
+
+    # pylint: disable=unused-argument, arguments-differ
+    def retrieve(self, request, pk=None):
+        """Get a scan job."""
```
Comment (chambridge) [inline quipucords/api/scan/view.py]: doc strings ...

File: quipucords/api/source/serializer.py [hunk 11]
```diff
@@ -52,7 +52,7 @@ def display_value(self, instance):
 
 
 class SourceOptionsSerializer(NotEmptySerializer):
-    """Serializer for the ScanOptions model."""
+    """Serializer for the SourceOptionsSerializer model."""
```
Comment (chambridge) [inline quipucords/api/source/serializer.py]: Serializer for the SourceOptions model.

**Comment by kholdaway** [comment]
Addressed all comments.

**Pull request merged**